src/streamdiffusion/stream_parameter_updater.py

```diff
@@ -1206,65 +1206,69 @@ def _update_ipadapter_config(self, desired_config: Dict[str, Any]) -> None:
-        if 'weight_type' in desired_config:
+        if 'weight_type' in desired_config and desired_config['weight_type'] is not None:
             weight_type = desired_config['weight_type']
-            try:
-                setattr(self.stream, 'ipadapter_weight_type', weight_type)
-            except Exception:
-                pass
-            # For PyTorch UNet, immediately apply a per-layer scale vector so layers reflect selection types
-            try:
-                is_tensorrt_engine = hasattr(self.stream.unet, 'engine') and hasattr(self.stream.unet, 'stream')
-                if not is_tensorrt_engine and hasattr(ipadapter_pipeline, 'ipadapter') and ipadapter_pipeline.ipadapter is not None:
-                    # Compute per-layer vector using Diffusers_IPAdapter helper
-                    from diffusers_ipadapter.ip_adapter.attention_processor import build_layer_weights
-                    # Count installed IP layers by scanning processors with _ip_layer_index
-                    ip_procs = [p for p in self.stream.pipe.unet.attn_processors.values() if hasattr(p, "_ip_layer_index")]
-                    num_layers = len(ip_procs)
-                    base_weight = float(getattr(self.stream, 'ipadapter_scale', getattr(ipadapter_pipeline, 'scale', 1.0)))
-                    weights = build_layer_weights(num_layers, base_weight, weight_type)
-                    # If None, keep uniform base scale; else set per-layer vector
-                    if weights is not None:
-                        ipadapter_pipeline.ipadapter.set_scale(weights)
-                    else:
-                        ipadapter_pipeline.ipadapter.set_scale(base_weight)
-            except Exception:
-                # Do not add fallback mechanisms
-                raise
+            # Update IPAdapter instance
+            if hasattr(self.stream, 'ipadapter'):
+                setattr(self.stream.ipadapter, 'weight_type', weight_type)
+                
+                # For PyTorch UNet, immediately apply a per-layer scale vector so layers reflect selection types
+                try:
+                    is_tensorrt_engine = hasattr(self.stream.unet, 'engine') and hasattr(self.stream.unet, 'stream')
+                    if not is_tensorrt_engine:
+                        # Compute per-layer vector using Diffusers_IPAdapter helper
+                        from diffusers_ipadapter.ip_adapter.attention_processor import build_layer_weights
+                        # Count installed IP layers by scanning processors with _ip_layer_index
+                        ip_procs = [p for p in self.stream.pipe.unet.attn_processors.values() if hasattr(p, "_ip_layer_index")]
+                        num_layers = len(ip_procs)
+                        # Get base weight from IPAdapter instance
+                        base_weight = float(getattr(self.stream.ipadapter, 'scale', 1.0))
```

Comment:
Related to the other comment, this should always work instead:

```suggestion
                        base_weight = self.stream.ipadapter.scale
```

- `getattr` is unnecessary as the field should always be defined with a default value defined in the constructor
- converting to float should also be unnecessary, the `ipadapter` itself should ensure the field can only be set to a `float` (don't even need to be super strict, just relying on typing should be enough)